src/testing_util/openssl_assertions.h

```diff
@@ -0,0 +1,138 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef KMSENGINE_TESTING_UTIL_OPENSSL_ASSERTIONS_H_
+#define KMSENGINE_TESTING_UTIL_OPENSSL_ASSERTIONS_H_
+
+#include <openssl/err.h>
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "absl/strings/string_view.h"
+#include "absl/strings/str_cat.h"
+#include "absl/strings/str_format.h"
+
+namespace kmsengine {
+namespace testing_util {
+
+// Helper function for formatting the custom assertion message.
+inline testing::AssertionResult MakeFormattedAssertionFailure(
+    absl::string_view message,
+    absl::string_view expected,
+    absl::string_view actual) {
+  // Indentation for "Expected" and "Actual" follows googlemock ASSERT_* and
+  // EXPECT_* formatting.
+  return testing::AssertionFailure() << absl::StrFormat(
+      "%s\n"
+      "         Expected: %s\n"
+      "           Actual: %s", message, expected, actual);
+}
+
+// Returns `testing::AssertionSuccess` if `openssl_return` is true. Otherwise,
+// returns `testing::AssertionFailure` with a formatted message containing the
+// first OpenSSL error in the OpenSSL error queue.
+//
+// Defined as a predicate instead of a matcher since matchers must be purely
+// functional, but `ERR_get_error_line_data` has OpenSSL-internal side-effects.
+testing::AssertionResult IsOpenSslSuccess(const char* expr,
+                                          bool openssl_return) {
+  if (openssl_return) return testing::AssertionSuccess();
+
+  const char *file, *data;
+  int line, flags;
+  auto err = ERR_get_error_line_data(&file, &line, &data, &flags);
+
+  // If err == 0, then no error was available in the OpenSSL error queue.
+  if (err) {
+    auto message = absl::StrFormat("%s unexpectedly returned false and "
```

Comment:
Nit: Rather than declaring single use variable, just incorporate the values into the `MakeFormattedAssertionFailure` expression.